File: backend/app/models/chapter_draft_attempt.py
"""章节草稿尝试模型。"""
from sqlalchemy import Column, String, Text, Integer, DateTime, ForeignKey, JSON
from sqlalchemy.sql import func
from app.database import Base
import uuid


class ChapterDraftAttempt(Base):
    """记录章节生成、批量生成与重写过程中的草稿尝试。"""
    __tablename__ = "chapter_draft_attempts"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    project_id = Column(String(36), ForeignKey("projects.id", ondelete="CASCADE"), nullable=False, comment="项目ID")
    chapter_id = Column(String(36), ForeignKey("chapters.id", ondelete="SET NULL"), nullable=True, comment="章节ID")
    batch_task_id = Column(String(36), ForeignKey("batch_generation_tasks.id", ondelete="SET NULL"), nullable=True, comment="批量任务ID")
    source = Column(String(40), nullable=False, default="chapter", comment="来源：chapter/batch/regenerate")
    attempt_state = Column(String(40), nullable=False, default="candidate", comment="尝试状态")
    quality_gate_action = Column(String(40), nullable=True, comment="质量门动作")
    quality_gate_decision = Column(String(40), nullable=True, comment="质量门决策")
    word_count = Column(Integer, nullable=False, default=0, comment="草稿字数")
    summary_preview = Column(Text, nullable=True, comment="摘要预览")
    content_preview = Column(Text, nullable=True, comment="内容预览")
    quality_metrics = Column(JSON, nullable=True, comment="质量指标")
    repair_payload = Column(JSON, nullable=True, comment="修复载荷")
    created_at = Column(DateTime, server_default=func.now(), comment="创建时间")

    def __repr__(self):
        return (
            f"<ChapterDraftAttempt(id={self.id}, chapter_id={self.chapter_id}, "
            f"attempt_state={self.attempt_state})>"
        )

File: backend-rs/src/models/chapter_draft_attempt.rs
use chrono::NaiveDateTime;
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Serialize, Deserialize)]
#[sea_orm(table_name = "chapter_draft_attempts")]
pub struct Model {
    #[sea_orm(primary_key, auto_increment = false)]
    pub id: String,
    pub project_id: String,
    pub chapter_id: Option<String>,
    pub batch_task_id: Option<String>,
    pub source: String,
    pub attempt_state: String,
    pub quality_gate_action: Option<String>,
    pub quality_gate_decision: Option<String>,
    pub word_count: i32,
    #[sea_orm(column_type = "Text", nullable)]
    pub summary_preview: Option<String>,
    #[sea_orm(column_type = "Text", nullable)]
    pub content_preview: Option<String>,
    pub quality_metrics: Option<serde_json::Value>,
    pub repair_payload: Option<serde_json::Value>,
    pub created_at: Option<NaiveDateTime>,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}